pkg/operation/seed/seed.go

```diff
@@ -1660,3 +1664,43 @@ func cleanupOrphanExposureClassHandlerResources(ctx context.Context, c client.Cl
 
 	return nil
 }
+
+// resizeOrDeleteLokiDataVolumeIfStorageNotTheSame updates the garden Loki PVC if passed storage value is not the same as the current one.
+// Caution: If the passed storage capacity is less than the current one the existing PVC and its PV will be deleted.
+func resizeOrDeleteLokiDataVolumeIfStorageNotTheSame(ctx context.Context, k8sClient client.Client, newStorageQuantity resource.Quantity, log logrus.FieldLogger) error {
+	// Check if we need resizing
+	pvc := &corev1.PersistentVolumeClaim{}
+	if err := k8sClient.Get(ctx, kutil.Key(v1beta1constants.GardenNamespace, "loki-loki-0"), pvc); err != nil {
+		return client.IgnoreNotFound(err)
+	}
+
+	storageCmpResult := newStorageQuantity.Cmp(*pvc.Spec.Resources.Requests.Storage())
+	if storageCmpResult == 0 {
+		return nil
+	}
+
+	log.Infof("Scaling StatefulSet %s/%s to zero in order to detach pvc %q", v1beta1constants.GardenNamespace, v1beta1constants.StatefulSetNameLoki, pvc.Name)
```

Comment:
Nit: (for consistency) somewhere pvc is lowecase, somewhere - uppercase.

```suggestion
	log.Infof("Scaling StatefulSet %s/%s to zero in order to detach PVC %q", v1beta1constants.GardenNamespace, v1beta1constants.StatefulSetNameLoki, pvc.Name)
```